File: Trabalho flutter 2/flutter_application_2/ControleFinanceiro-main/app/src/main/java/br/edu/ifsuldeminas/mch/controlefinanceiro/model/Bill.java
package br.edu.ifsuldeminas.mch.controlefinanceiro.model;

import androidx.annotation.NonNull;

import java.io.Serializable;

public class Bill implements Serializable {

    private Integer id;
    private String name, description, date;
    private Double value;

    public Bill() {

    }

    public Bill(Integer id, String name, String description, Double value, String date) {
        this.id = id;
        setName(name);
        setDescription(description);
        setValue(value);
        setDate(date);
    }


    public Integer getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Double getValue() {
        return value;
    }

    public void setValue(Double value) {
        this.value = value;
    }

    @NonNull
    @Override
    public String toString() {
        return "R$" + this.getValue() + " - " + this.getName() + " - " + this.getDate();
    }
}

File: Trabalho flutter 2/flutter_application_2/lib/models/bill.dart
/// Modelo Bill (Conta a Pagar)
///
/// Migrado de: br.edu.ifsuldeminas.mch.controlefinanceiro.model.Bill (Java)
///
/// **Diferenças do Java:**
/// - No Java, usávamos getters/setters manuais
/// - Em Dart, usamos construtores nomeados e propriedades finais (imutabilidade)
/// - Melhor integração com JSON através de métodos toMap/fromMap
class Bill {
  final int? id;
  final String name;
  final String description;
  final double value;
  final String date; // Formato: dd/MM/yyyy

  Bill({
    this.id,
    required this.name,
    required this.description,
    required this.value,
    required this.date,
  });

  /// Converte Bill para Map (usado pelo SQLite)
  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'value': value,
      'date': date,
    };
  }

  /// Cria Bill a partir de Map (vindo do SQLite)
  factory Bill.fromMap(Map<String, dynamic> map) {
    return Bill(
      id: map['id'] as int?,
      name: map['name'] as String,
      description: map['description'] as String,
      value: map['value'] as double,
      date: map['date'] as String,
    );
  }

  /// Cria cópia com modificações (útil para edições)
  Bill copyWith({
    int? id,
    String? name,
    String? description,
    double? value,
    String? date,
  }) {
    return Bill(
      id: id ?? this.id,
      name: name ?? this.name,
      description: description ?? this.description,
      value: value ?? this.value,
      date: date ?? this.date,
    );
  }

  @override
  String toString() {
    return 'R\$ ${value.toStringAsFixed(2)} - $name - $date';
  }
}
